dojo-dungeon/src/dojo/monster/MyImp.java

```diff
@@ -18,8 +19,9 @@ public class MyImp {
    *
    * @param entity the entity to create
    * @param isInitialized if the entity was already initialized
+   * @param currentRoom the current room the entity is in
    */
-  public MyImp(Entity entity, boolean isInitialized) {
+  public MyImp(Entity entity, boolean isInitialized, Room currentRoom) {
```

Comment:
Das sieht für mich aus wie ein klassischer Fall von Feature-Neid. Die Klasse `MyImp` hat keinerlei Attribute und arbeitet nur auf dem Parameter `entity`. Ich habe keine Idee, wo Du mit dieser Klasse hin willst (die restlichen Methoden stehen auf "TODO"), aber das sieht für mich eher danach aus, dass Du eine Klasse mit einer statischen Methode haben möchtest, die Dir Deine Entität fertig konfiguriert (statt einer Klasse mit Konstruktor)?

Reply:
Das ist in der Tat etwas unschön. Ich ändere das in mehrere (statische) Methoden. Eine, um die Grundeigenschaften des Imps zu setzen (das habe ich hier etwas oberflächlich nur "Initialisierung" genannt), und mehrere andere, die von den Studis implementiert werden sollen, entsprechend der Java-Docs.